Tests/SKCoreTests/SplitShellCommandTests.swift

```diff
@@ -0,0 +1,119 @@
+    assertEscapedCommand("abc def", ["abc", "def"])
+    assertEscapedCommand("abc    def", ["abc", "def"])
+  }
+
+  func testSplitShellEscapedCommandDoubleQuotes() {
+    assertEscapedCommand("\"", [""])
+    assertEscapedCommand(#""a"#, ["a"])
+    assertEscapedCommand("\"\"", [""])
+    assertEscapedCommand(#""a""#, ["a"])
+    assertEscapedCommand(#""a\"""#, [#"a""#])
+    assertEscapedCommand(#""a b c ""#, ["a b c "])
+    assertEscapedCommand(#""a " "#, ["a "])
+    assertEscapedCommand(#""a " b"#, ["a ", "b"])
+    assertEscapedCommand(#""a "b"#, ["a b"])
+    assertEscapedCommand(#"a"x ""b"#, ["ax b"], windows: [#"ax "b"#])
+
+    assertEscapedCommand(#""a"bcd"ef""""g""#, ["abcdefg"], windows: [#"abcdef""g"#])
+  }
+
+  func testSplitShellEscapedCommandSingleQuotes() {
+    assertEscapedCommand("'", [""], windows: ["'"])
+    assertEscapedCommand("'a", ["a"], windows: ["'a"])
+    assertEscapedCommand("''", [""], windows: ["''"])
+    assertEscapedCommand("'a'", ["a"], windows: ["'a'"])
+    assertEscapedCommand(#"'a\"'"#, [#"a\""#], windows: [#"'a"'"#])
+    assertEscapedCommand(#"'a b c '"#, ["a b c "], windows: ["'a", "b", "c", "'"])
+    assertEscapedCommand(#"'a ' "#, ["a "], windows: ["'a", "'"])
+    assertEscapedCommand(#"'a ' b"#, ["a ", "b"], windows: ["'a", "'", "b"])
+    assertEscapedCommand(#"'a 'b"#, ["a b"], windows: ["'a", "'b"])
+    assertEscapedCommand(#"a'x ''b"#, ["ax b"], windows: ["a'x", "''b"])
+  }
+
+  func testSplitShellEscapedCommandBackslash() {
+    assertEscapedCommand(#"a\\"#, [#"a\"#], windows: [#"a\\"#])
+    assertEscapedCommand(#"a'\b "c"'"#, ["a\\b \"c\""], windows: [#"a'\b"#, #"c'"#])
+
+    assertEscapedCommand(#"\""#, ["\""])
+    assertEscapedCommand(#"\\""#, [#"\"#])
+    assertEscapedCommand(#"\\\""#, [#"\""#])
+    assertEscapedCommand(#"\\ "#, [#"\"#], windows: [#"\\"#])
+    assertEscapedCommand(#"\\\ "#, [#"\ "#], windows: [#"\\\"#])
+  }
+
+  func testSplitShellEscapedCommandWindowsCommand() {
+    assertEscapedCommand(#"C:\swift.exe"#, [#"C:swift.exe"#], windows: [#"C:\swift.exe"#], initialCommandName: true)
+    assertEscapedCommand(
+      #"C:\ swift.exe"#,
+      [#"C: swift.exe"#],
+      windows: [#"C:\"#, #"swift.exe"#],
+      initialCommandName: true
+    )
+    assertEscapedCommand(
+      #"C:\ swift.exe"#,
+      [#"C: swift.exe"#],
+      windows: [#"C:\"#, #"swift.exe"#],
+      initialCommandName: false
+    )
+    assertEscapedCommand(#"C:\"swift.exe""#, [#"C:"swift.exe"#], windows: [#"C:\swift.exe"#], initialCommandName: true)
+    assertEscapedCommand(#"C:\"swift.exe""#, [#"C:"swift.exe"#], windows: [#"C:"swift.exe"#], initialCommandName: false)
+  }
+
+  func testSplitShellEscapedCommandWindowsTwoDoubleQuotes() {
+    assertEscapedCommand(#"" test with "" quote""#, [" test with  quote"], windows: [#" test with " quote"#])
+    assertEscapedCommand(#"" test with "" quote""#, [" test with  quote"], windows: [#" test with " quote"#])
```

Comment:
Identical?

Reply:
No, Unix has `with  quote`, Windows has `with " quote` (double `""` results in an escaped `"` on Windows and an empty String on Unix)